back-end/utils/top-layer/UserUtil.js

```diff
@@ -365,44 +365,55 @@ const getLengthUserTransactionsHistoryForRedisM5RU = (email, filters) => {
 /**
  * Perform buy action for the function proceedTransaction
  * @param {import(".prisma/client").User} user
- * @param {number} totalPendingPrice
+ * @param {number} totalCashChange
  * @param {string} companyCode
+ * @param {number} quantity
+ * @param {number} recentPrice 
  * @returns
  */
  const buyShareEvent = async (
   user,
-  totalPendingPrice,
+  totalCashChange,
   companyCode,
   quantity,
   recentPrice
 ) => {
   try {
-    if (user.cash < totalPendingPrice) {
+    if (user.cash < totalCashChange) {
       return new Error(
         "The user does not have enough money to buy the pending stock"
       );
     }
+    // Update user cash
```

Comment:
Split similar parts of buy and sell to separate functions. For example: updateUserCash(user, totalChange).